tests/core/test_index.py

```diff
@@ -558,6 +558,17 @@ def test_eval_operators(index_data, op, value, expected):
     assert result == expected
 
 
+def test_eval_operators_type_safety():
+    # gh66
+    ind = IndexBase(column="col", index_dct={1234: ["part"]}, dtype=pa.int64())
+    with pytest.raises(TypeError):
```

Comment:
please assert the exception message here, otherwise we might catch some other ktk bug instead